### PR TITLE
[Feature] Checkbox/CheckboxGroup: Add heading role and list position

### PR DESCRIPTION
**Pull request by narin**
## Description of Change

This PR contains accessibility improvements as suggested by @brea11y.
- Adds `heading` accessibility role to Header when provided to Checkbox or Checkbox group
- Add convenience hook `useA11yListPosition` to return `listPosition` translation when provided position and total length
  - Updated SegmentedControl to use this hook
- Adds `accessibilityValue` to Checkbox items within Checkbox group to allow screen reader to read out position within list

### Screenshots/Video

https://github.com/user-attachments/assets/d2766d7a-6758-422f-b5c4-4dd51228d46d

#### A11y label videos

https://github.com/user-attachments/assets/c4d3a27d-f279-47a5-8eb9-5d84b6c5a19b

https://github.com/user-attachments/assets/d615f978-70ac-4d50-bc18-41e1222d03e7

## Testing

- [x] Tested on iOS 
- [x] Tested on Android 
- [x] Tested on Web

## PR Checklist
Code reviewer validation:
- General
	- [x] PR is linked to ticket(s)
	- [x] PR has `changelog` label applied if it's to be included in the changelog
	- [x] Acceptance criteria: 
		- All satisfied _or_
		- Documented reason for not being performed _or_
		- Split to separate ticket and ticket is linked by relevant AC(s)
	- [x] Above PR sections adequately filled out
	- [x] If any breaking changes, in accordance with the [pre-1.0.0 versioning guidelines](https://github.com/department-of-veterans-affairs/va-mobile-library#versioning-policy): a CU ticket has been created for the VA Mobile App detailing necessary adjustments with the package version that will be published by this ticket
- Code
	- [x] Tests are included if appropriate (or split to separate ticket)
	- [x] New functions have proper TSDoc annotations

## Publish

If changes warrant a new version [per the versioning guidelines](https://github.com/department-of-veterans-affairs/va-mobile-library/blob/main/documentation/versioning.md) and the PR is approved and ready to merge:
- [ ] Merge `main` into branch
- [ ] Merge branch to `main`
- [ ] Verify that [Check Component Integrations](https://github.com/department-of-veterans-affairs/va-mobile-library/actions/workflows/check-component-integrations.yml) workflow ran successfully
- [ ] [Publish new version](https://github.com/department-of-veterans-affairs/va-mobile-library/actions/workflows/publish.yml) after merging

### CODE REVIEW

File: packages/components/src/components/Checkbox/Checkbox.tsx [hunk 1]
```diff
@@ -130,6 +131,7 @@ export const Checkbox: FC<CheckboxProps> = ({
         <Pressable
           onPress={onPress}
           style={tile ? tileStyle : pressableBaseStyle}
+          accessibilityValue={accessibilityValue}
```
Comment (TimRoe) [inline packages/components/src/components/Checkbox/Checkbox.tsx:134]: Could be updated to `aria-valuetext` to align with newer/more web-like RN props.

File: packages/components/src/utils/hooks/useA11yListPosition.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+/** Returns list position used for accessibilityValues */
+export function useA11yListPosition(
+  position: number,
+  total: number,
+): AccessibilityValue {
+  const { t } = useTranslation()
+
+  return {
+    text: t('listPosition', {
+      position,
```
Comment (TimRoe) [inline packages/components/src/utils/hooks/useA11yListPosition.ts:13]: Should the +1 be baked into the hook? And rename the argument from `position` to `index` to set the expectation it counts from 0.

For user facing counting, the expectation is it'd never start from 0 so seems reasonable to me to roll that into the utility and only correct as a one-off with -1 if for some reason we were working with values that didn't start from 0.

File: packages/components/src/components/Checkbox/Checkbox.tsx (hunk 1 above)
Comment (TimRoe) [reply packages/components/src/components/Checkbox/Checkbox.tsx:134]: Nvm, couldn't here using the more general RN typing without updating that to string.

File: packages/components/src/components/CheckboxGroup/CheckboxGroup.tsx [hunk 3]
```diff
@@ -142,10 +142,16 @@ export const CheckboxGroup: FC<CheckboxGroupProps> = ({
...
+          const accessibilityValue = useA11yListPosition(
+            index + 1,
+            items.length,
+          )
+
           return (
             <Fragment key={`checkbox-group-item-${index}`}>
               <Checkbox
                 label={item}
+                accessibilityValue={accessibilityValue}
```
Comment (TimRoe) [inline packages/components/src/components/CheckboxGroup/CheckboxGroup.tsx:154]: Could be updated to `aria-valuetext` to align with newer/more web-like RN props.

File: packages/components/src/components/SegmentedControl/SegmentedControl.tsx [hunk 4]
```diff
@@ -83,12 +85,7 @@ export const SegmentedControl: FC<SegmentedControlProps> = ({
     const accessibilityLabel = a11yLabels
       ? a11yLabels[index] || labels[index]
       : labels[index]
-    const accessibilityValue = {
-      text: t('listPosition', {
-        position: index + 1,
-        total: labels.length,
-      }),
-    }
+    const accessibilityValue = useA11yListPosition(index + 1, labels.length)
```
Comment (TimRoe) [inline packages/components/src/components/SegmentedControl/SegmentedControl.tsx:88]: Could potentially be in-line without wrapping now, eliminating the variable.

File: packages/components/src/components/Checkbox/Checkbox.tsx [hunk 5]
```diff
@@ -130,6 +131,7 @@ export const Checkbox: FC<CheckboxProps> = ({
         <Pressable
```
Comment (TimRoe) [inline packages/components/src/components/Checkbox/Checkbox.tsx:131]: Does it cause issues with this being on the Pressable? Namely, do the a11yLabel overrides still work correctly when they're now nested a layer lower than the a11yValue? The PR video only highlights options 3/4 which don't have a label override.

**Review by TimRoe — CHANGES_REQUESTED** [review]
Mainly requesting changes as I think these do warrant adding to the unit tests for Checkbox/CheckboxGroup (validating header role and a11yValue) and the general ticket for that has already been worked.

Additionally:
- Recommend moving the +1 into the `useA11yListPosition` hook
- Unclear if the a11yValue being at a different level from a11yLabel causes issues with both being read as expected based on PR video

Beyond that, some minor suggestions.

**Comment by narin** [comment]
Ready for another review @TimRoe:
- Updated list position hook to accept index and add + 1 to calculate position
- Alphabetized exports in accessibility.ts
- Updated components to use hook inline instead of declaring separately
- Updated Checkbox and SegmentedControl to use `aria-valuetext` for list position

File: packages/components/src/components/Checkbox/Checkbox.tsx (hunk 5 above)
Comment (narin) [reply packages/components/src/components/Checkbox/Checkbox.tsx]: Confirmed that the a11y label overrides still work correctly with voiceover

**Review by TimRoe — CHANGES_REQUESTED** [review]
Changes look good, but noticed a thing I didn't before: since this is a hook, we should be instantiating it at the top with a `const a11yListPosition = useA11yListPosition()` or whatever to be certain we aren't conditionally changing the number of hooks, right? I don't think is an issue to these particular use cases, but think is the general best practice to avoid situations like that recurring Android crash in flagship.

**Comment by narin** [comment]
- Deleted useA11yListPosition hook and reverted to using `listPosition` translation directly since we're now using `aria-valuetext`
- Added unit tests to check for heading role and accessibility value
- Fix bug where Talkback wasn't reading a11y labels for Label and Description but combining aria-labels and setting on Pressable

File: packages/components/src/components/Checkbox/Checkbox.tsx [hunk 6]
```diff
@@ -20,7 +20,8 @@ import {
 } from '../shared/FormText'
 import { Icon, IconProps } from '../Icon/Icon'
 import { Spacer } from '../Spacer/Spacer'
-import { useTheme } from '../../utils'
+import { getA11yLabel, useTheme } from '../../utils'
+import { useTranslation } from 'react-i18next'
```
Comment (TimRoe) [inline packages/components/src/components/Checkbox/Checkbox.tsx:24]: Nitpick: move up to library imports.

File: packages/components/src/components/CheckboxGroup/CheckboxGroup.tsx [hunk 7]
```diff
@@ -11,7 +11,8 @@ import {
   TextWithA11y,
 } from '../../types'
 import { Spacer } from '../Spacer/Spacer'
-import { useA11yListPosition, useTheme } from '../../utils'
+import { useTheme } from '../../utils'
+import { useTranslation } from 'react-i18next'
```
Comment (TimRoe) [inline packages/components/src/components/CheckboxGroup/CheckboxGroup.tsx:15]: Same nitpick as other file

File: packages/components/src/components/SegmentedControl/SegmentedControl.tsx [hunk 8]
```diff
@@ -4,11 +4,8 @@ import React, { FC, useEffect } from 'react'
 import styled from 'styled-components/native'
 
 import { ComponentWrapper } from '../../wrapper'
-import {
-  PressableOpacityStyle,
-  useA11yListPosition,
-  useTheme,
-} from '../../utils'
+import { PressableOpacityStyle, useTheme } from '../../utils'
+import { useTranslation } from 'react-i18next'
```
Comment (TimRoe) [inline packages/components/src/components/SegmentedControl/SegmentedControl.tsx:8]: Same nitpick as other files

File: packages/components/src/components/SegmentedControl/SegmentedControl.tsx [hunk 9]
```diff
@@ -103,6 +97,11 @@ export const SegmentedControl: FC<SegmentedControlProps> = ({
       textAlign: 'center',
     }
 
+    const a11yListPosition = t('listPosition', {
```
Comment (TimRoe) [inline packages/components/src/components/SegmentedControl/SegmentedControl.tsx:100]: Nitpicky, but old position of setting the const alongside the a11yLabel made more sense readability-wise

**Review by TimRoe — APPROVED** [review]
Approved with a few lingering nitpicks about import/const positioning.

**Pull request merged**